src/providers/details/tabs/Hosts/components/utils/onSaveHost.ts

```diff
@@ -0,0 +1,156 @@
+import { Base64 } from 'js-base64';
+
+import {
+  type IoK8sApiCoreV1Secret,
+  type NetworkAdapters,
+  SecretModel,
+  type V1beta1Host,
+  type V1beta1Provider,
+} from '@kubev2v/types';
+import { k8sGet } from '@openshift-console/dynamic-plugin-sdk';
+
+import { createHost } from './createHost';
+import { createHostSecret } from './createHostSecret';
+import { patchHost } from './patchHost';
+import { patchHostSecretOwner } from './patchHostSecretOwner';
+import { patchSecret } from './patchSecret';
+import type { InventoryHostNetworkTriple } from './types';
+
+type OnSaveHostParams = {
+  provider: V1beta1Provider;
+  hostPairs: InventoryHostNetworkTriple[];
+  network: NetworkAdapters;
+  user?: string;
+  passwd?: string;
+};
+
+const processHostSecretPair = async (
+  provider: V1beta1Provider,
+  hostPair: InventoryHostNetworkTriple,
+  ipAddress: string,
+  encodedUser: string | undefined,
+  encodedPassword: string | undefined,
+  encodedProvider: string,
+  encodedIpAddress: string,
+) => {
+  const { host, inventory } = hostPair;
+
+  if (host?.metadata?.name) {
+    // Host already set, update network in the host and secret
+    const { name: secretName, namespace: secretNamespace } = host?.spec?.secret ?? {};
+
+    const secretData = await k8sGet<IoK8sApiCoreV1Secret>({
+      model: SecretModel,
+      secretName,
+      secretNamespace,
+    });
+
+    await patchHost(host, ipAddress);
+    await patchSecret(secretData, encodedIpAddress, encodedUser, encodedPassword);
+  } else {
+    // Create a new host and secret pair
+
+    // Create a Secret
+    const secretData = {
+      apiVersion: 'v1',
+      data: {
+        ip: encodedIpAddress,
+        password: encodedPassword,
+        provider: encodedProvider,
+        user: encodedUser,
+      },
+      kind: 'Secret',
+      metadata: {
+        generateName: `${provider?.metadata?.name}-${inventory.id}-`,
+        labels: {
+          createdForResource: inventory.id,
+          createdForResourceType: 'hosts',
+        },
+        namespace: provider?.metadata?.namespace,
+      },
+      type: 'Opaque',
+    };
+    const createdSecret = await createHostSecret(secretData as IoK8sApiCoreV1Secret);
+
+    // Create Host
+    const newHostData = {
+      apiVersion: 'forklift.konveyor.io/v1beta1',
+      kind: 'Host',
+      metadata: {
+        name: `${provider?.metadata?.name}-${inventory.id}-config`,
+        namespace: provider?.metadata?.namespace,
+        ownerReferences: [
+          {
+            apiVersion: 'forklift.konveyor.io/v1beta1',
+            kind: 'Provider',
+            name: provider?.metadata?.name,
+            uid: provider?.metadata?.uid,
+          },
+        ],
+      },
+      spec: {
+        id: inventory.id,
+        ipAddress,
+        provider: {
+          name: provider?.metadata?.name,
```

Comment:
we have selectros for many of those fields, name, namespace, etc. please check and add missing ones